### PR TITLE
Pv2 1569 remove redundant earnings, required payments and funding source event data after submission succeeded or failed

### PR DESCRIPTION
**Pull request by abroodie**

### CODE REVIEW

File: src/SFA.DAS.Payments.AcceptanceTests.EndToEnd/Steps/EndToEndSteps.cs [hunk 1]
```diff
@@ -222,10 +260,41 @@ public void GivenPriceDetailsAsFollows(Table table)
...
+            var dcHelper = Scope.Resolve<IDcHelper>();
+            await dcHelper.SendIlrSubmissionEvent(TestSession.Provider.Ukprn, CurrentCollectionPeriod.AcademicYear,
+                CurrentCollectionPeriod.Period,
+                TestSession.Provider.JobId, true).ConfigureAwait(false);
+        }
+
+        [Then(@"the data for the previous submission should be removed")]
+        public void ThenTheDataForThePreviousSubmissionShouldBeRemoved()
+        {
+            ScenarioContext.Current.Pending();
```
Comment (samuel-dev-dba) [inline src/SFA.DAS.Payments.AcceptanceTests.EndToEnd/Steps/EndToEndSteps.cs]: This step should be implemented or removed

File: src/SFA.DAS.Payments.AcceptanceTests.EndToEnd/Tests/Levy Submission Events/One Levy Learner, levy available, finishes on time with submission events PV2-1395.feature [hunk 2]
```diff
@@ -1,88 +0,0 @@
-﻿@ignore - at the moment, there is an issue in the 1395 branch with the first message returning and hence the second one doesn't get consumed.
```
Comment (samuel-dev-dba) [inline src/SFA.DAS.Payments.AcceptanceTests.EndToEnd/Tests/Levy Submission Events/One Levy Learner, levy available, finishes on time with submission events PV2-1395.feature]: Why is this spec removed

File: src/SFA.DAS.Payments.AcceptanceTests.EndToEnd/Tests/Submission Events/PV2-1395 - Successful submission removes previous submission payments.cs [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+//      SpecFlow Version:2.4.0.0
+//      SpecFlow Generator Version:2.4.0.0
+// 
+//      Changes to this file may cause incorrect behavior and will be lost if
+//      the code is regenerated.
+//  </auto-generated>
+// ------------------------------------------------------------------------------
+#region Designer generated code
+#pragma warning disable
+namespace SFA.DAS.Payments.AcceptanceTests.EndToEnd.Tests.SubmissionEvents
```
Comment (samuel-dev-dba) [inline src/SFA.DAS.Payments.AcceptanceTests.EndToEnd/Tests/Submission Events/PV2-1395 - Successful submission removes previous submission payments.cs]: We don't merge auto-generated specflow .cs files this should be ignored.

File: src/SFA.DAS.Payments.AcceptanceTests.EndToEnd/Tests/Submission Events/PV2-1396 - Failed submission job removes payments for failed job.cs [hunk 4]
```diff
@@ -0,0 +1,101 @@
+﻿// ------------------------------------------------------------------------------
+//  <auto-generated>
```
Comment (samuel-dev-dba) [inline src/SFA.DAS.Payments.AcceptanceTests.EndToEnd/Tests/Submission Events/PV2-1396 - Failed submission job removes payments for failed job.cs]: We don't merge auto-generated specflow .cs files this should be ignored.

**Comment by ESFA-Pawan** [comment]
This is in prod now.

File: src/SFA.DAS.Payments.AcceptanceTests.EndToEnd/Tests/Submission Events/PV2-1396 - Failed submission job removes payments for failed job.cs (hunk 4 above)
Comment (abroodie) [reply src/SFA.DAS.Payments.AcceptanceTests.EndToEnd/Tests/Submission Events/PV2-1396 - Failed submission job removes payments for failed job.cs]: done

File: src/SFA.DAS.Payments.AcceptanceTests.EndToEnd/Tests/Submission Events/PV2-1395 - Successful submission removes previous submission payments.cs (hunk 3 above)
Comment (abroodie) [reply src/SFA.DAS.Payments.AcceptanceTests.EndToEnd/Tests/Submission Events/PV2-1395 - Successful submission removes previous submission payments.cs]: done

**Pull request merged**